buckets/redis/bucket.go

```diff
@@ -67,6 +67,19 @@ func (bf *bucketFactory) Init(cfg *pbconfig.ServiceConfig) {
 	if bf.client == nil {
 		bf.connectToRedisLocked()
 	}
+
+	if err := bf.cleanStaleBuckets() ; err != nil {
```

Comment:
We effectively lock the entire server for that entire time, so it's something to keep in mind.